### PR TITLE
MINOR: Replace test usages of ClientUtils.fetchTopicMetadata with BaseRequestTest

### PR DESCRIPTION
**Pull request by ijuma**
For tests that are not testing the old consumers functionality. As part of this,
consolidate `TopicMetadataTest` into `MetadataRequestTest`. Finally,
remove `ProducerBounceTest` which has no tests left in it.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala [hunk 1]
```diff
@@ -235,8 +280,81 @@ class MetadataRequestTest extends BaseRequestTest {
...
+        controllerMetadataResponse = Some(metadataResponse)
+        metadataResponse.brokers.size == expectedBrokersCount
+      }, s"Expected $expectedBrokersCount brokers, but there are ${controllerMetadataResponse.get.brokers.size} " +
+        "according to the Controller")
+
+      val brokersInController = controllerMetadataResponse.get.brokers.asScala.toSeq.sortBy(_.id)
+
+      // Assert that metadata is propagated correctly
+      servers.filter(_.brokerState.currentState != NotRunning.state).foreach { broker =>
+        println(s"broker ${broker.brokerState}")
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala]: Probably unintended

**Review by hachikuji — APPROVED** [review]
LGTM. Just one trivial comment.

File: core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala]: Good catch! Fixed.

**Comment by ijuma** [comment]
There was a single unrelated test failure across both jobs. Merging to trunk and 2.0.

**Pull request merged**